openedx/core/djangoapps/embargo/api.py

```diff
@@ -7,34 +7,53 @@
 """
 
 import logging
+from typing import List, Optional
 
 from django.conf import settings
 from django.core.cache import cache
-from ipware.ip import get_client_ip
+from opaque_keys.edx.keys import CourseKey
 from rest_framework import status
+from rest_framework.request import Request
 from rest_framework.response import Response
 
 from common.djangoapps.student.auth import has_course_author_access
+from openedx.core import types
 from openedx.core.djangoapps.geoinfo.api import country_code_from_ip
+from openedx.core.djangoapps.util import ip
 
 from .models import CountryAccessRule, RestrictedCourse
 
 log = logging.getLogger(__name__)
 
 
-def redirect_if_blocked(course_key, access_point='enrollment', **kwargs):
-    """Redirect if the user does not have access to the course. In case of blocked if access_point
-    is not enrollment and course has enabled is_disabled_access_check then user can view that course.
+def redirect_if_blocked(
+    request: Request,
+    course_key: CourseKey,
+    access_point: str = 'enrollment',
+    user: Optional[types.User] = None,
+) -> Optional[str]:
+    """
+    Redirect if the user does not have access to the course.
 
-    Arguments:
-        course_key (CourseKey): Location of the course the user is trying to access.
+    Even if the user would normally be blocked, if the given access_point is 'courseware' and the course has enabled
+    the `is_disabled_access_check` flag, then the user can still view that course.
 
-    Keyword Arguments:
-        Same as `check_course_access` and `message_url_path`
+    Arguments:
+        request: The current request to be checked.
+        course_key: Location of the course the user is trying to access.
+        access_point: Type of page being accessed (e.g. 'courseware', 'enrollment', etc)
+        user: User to check for (uses request.user if None)
 
+    Returns:
+        If blocked, a URL path to a page explaining why the user was blocked. Else None.
     """
     if settings.FEATURES.get('EMBARGO'):
-        is_blocked = not check_course_access(course_key, **kwargs)
+        if ip.USE_LEGACY_IP.is_enabled():
+            client_ips = [ip.get_legacy_ip(request)]
+        else:
+            client_ips = ip.get_all_client_ips(request)
```

Comment:
I would prefer for `ip.get_all_client_ips` to abstract this logic for us and check `USE_LEGACY_IP` for us, but the annotations for this waffle switch are very clear that callers should check it - I _assume_ with the intention that new code doesn't need to consider this switch, but old code that is ported should.